python/ray/serve/tests/test_http_state.py

```diff
@@ -46,6 +44,13 @@ def mock_get_all_node_ids(all_nodes):
         yield
 
 
+@pytest.fixture()
+def mock_get_head_node_id(all_nodes):
+    with patch("ray.serve._private.http_state.get_head_node_id") as func:
+        func.return_value = HEAD_NODE_ID
+        yield
```

Comment:
don't need this if do my suggestion above :)

in general, if you have to mock something you're probably doing it wrong...